crypto/bio/md.c

```diff
@@ -119,14 +120,33 @@ static long md_ctrl(BIO *b, int cmd, long num, void *ptr) {
         BIO_set_init(b, 1);
       }
       break;
-    // OpenSSL implements these, but because we don't need them and cipher BIO
-    // is internal, we can fail loudly if they're called. If this case is hit,
-    // it likely means you're making a change that will require implementing
-    // these.
     case BIO_C_GET_MD:
+      GUARD_PTR(ptr);
+      if (BIO_get_init(b)) {
+        const EVP_MD **ppmd = ptr;
+        *ppmd = EVP_MD_CTX_md(ctx);
+      } else {
+        ret = 0;
```

Comment:
very minor nit: this line isn't covered by tests

Reply:
Added a minor new test!